server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -722,6 +731,13 @@ public <Request extends ActionRequest, Response extends ActionResponse> void reg
         return unmodifiableMap(actions.getRegistry());
     }
 
+    public DynamicActionRegistry getExtensionActions() {
```

Comment:
Why should extension actions be treated differently by callers of ActionModule? Seems like they are all still actions

Reply:
I've combined the two maps (immutable and dynamic) inside the registry so they're an implementation detail. I think that tries to address this where external to the module they can be treated the same.